plugins/linux-capture/CMakeLists.txt

```diff
@@ -1,101 +1,71 @@
 project(linux-capture)
 
+option(ENABLE_PIPEWIRE "Enable PipeWire support" ON)
+if(NOT ENABLE_PIPEWIRE)
+  message(STATUS "OBS: PipeWire support not enabled")
+endif()
+
 find_package(X11 REQUIRED)
-if(NOT X11_Xcomposite_FOUND)
-	message(STATUS "Xcomposite library not found, linux-capture plugin disabled")
-	return()
+if(NOT TARGET X11::Xcomposite)
+  message(FATAL_ERROR "OBS: Xcomposite library not found, plugin disabled")
 endif()
+find_package(XCB COMPONENTS XCB XFIXES RANDR SHM XINERAMA)
 
-find_package(XCB COMPONENTS XCB RANDR SHM XFIXES XINERAMA REQUIRED)
-find_package(X11_XCB REQUIRED)
+add_library(linux-capture MODULE)
+add_library(OBS::capture ALIAS linux-capture)
 
-set(linux-capture_INCLUDES
-	"${CMAKE_SOURCE_DIR}/libobs"
-	${X11_Xcomposite_INCLUDE_PATH}
-	${X11_X11_INCLUDE_PATH}
-	${XCB_INCLUDE_DIRS}
-)
+target_sources(
+  linux-capture
+  PRIVATE linux-capture.c
+          xcursor.c
+          xcursor.h
+          xcursor-xcb.c
+          xcursor-xcb.h
+          xhelpers.c
+          xhelpers.h
+          xshm-input.c
+          xcomposite-main.cpp
+          xcompcap-main.cpp
+          xcompcap-main.hpp
+          xcompcap-helper.cpp
+          xcompcap-helper.hpp)
 
-set(linux-capture_SOURCES
-	linux-capture.c
-	xcursor.c
-	xcursor-xcb.c
-	xhelpers.c
-	xshm-input.c
-	xcomposite-main.cpp
-	xcompcap-main.cpp
-	xcompcap-helper.cpp
-)
-set(linux-capture_HEADERS
-	xcursor.h
-	xcursor-xcb.h
-	xhelpers.h
-	xcompcap-main.hpp
-	xcompcap-helper.hpp
-)
+target_link_libraries(
+  linux-capture
+  PRIVATE OBS::libobs
+          OBS::obsglad
+          X11::X11
+          X11::Xfixes
+          X11::Xcomposite
+          XCB::XCB
+          XCB::XFIXES
+          XCB::RANDR
+          XCB::SHM
+          XCB::XINERAMA)
 
-set(linux-capture_LIBRARIES
-	libobs
-	glad
-	${X11_LIBRARIES}
-	${X11_Xfixes_LIB}
-	${X11_X11_LIB}
-	${X11_Xcomposite_LIB}
-	${XCB_LIBRARIES}
-)
+set_target_properties(linux-capture PROPERTIES FOLDER "plugins")
 
-option(ENABLE_PIPEWIRE "Enable PipeWire support" ON)
 if(ENABLE_PIPEWIRE)
-	find_package(PipeWire QUIET)
-	find_package(Gio QUIET)
+  find_package(PipeWire REQUIRED)
+  find_package(Gio REQUIRED)
```

Comment:
By putting `REQUIRED` here, CMake will automatically emit an error, setting them to `QUIET` would make CMake able to emit the following custom `FATAL_ERROR` messages.

*Otherwise the custom messages are useless.*

Reply:
Nevermind, I discover that `REQUIRED` don't stop CMake even if it doesn't find it